rasa/core/brokers/kafka.py

```diff
@@ -240,6 +241,9 @@ def _publish(self, event: Dict[Text, Any]) -> None:
                 )
             ]
 
+        # set stdout to non-blocking
+        fcntl.fcntl(1, fcntl.F_SETFL, 0)
```

Comment:
I think this won't work on Windows, it's only a solution for Unix systems, we'll need something universal for any OS.

Reply:
Also, we should probably seek all potential log locations with too much data that causes this issue, not only this module.

Reply:
Yeah, tests are already failing.

Reply:
> Also, we should probably seek all potential log locations with too much data that causes this issue, not only this module.
Only this particular log caused the issue. But I can future proof all logs with large data.

Reply:
🙏🏻

Reply:
> replace keys that could hold too much data with an optimized version of that, e.g. rasa_event, rasa_events, extraction_events etc.

Are we saying we can send out a reduced version instead? Like with less data?

Reply:
Something worth exploring based on how previous logs used to look like.